### PR TITLE
#170947547 Adds login feature

### PR DESCRIPTION
**Pull request by Baraka-Mugisha**
#### What does this PR do?
- create login API endpoint for Barefoot Nomad

#### Description of Task to be completed?
- create a login route at “api/v1/auth/login” `
- add login tests
- add login controllers
- add login routes

#### How should this be manually tested?
- clone the repo
- run `npm install` to install all dependencies
- run the server and go to the above route in POSTMAN

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#170947547](https://www.pivotaltracker.com/story/show/170947547)

#### Screenshots (if appropriate)
Screen Shot 2020-02-13 at 15 48 44
Screen Shot 2020-02-13 at 15 47 46

### CODE REVIEW

**Review by Ntare22 — DISMISSED** [review]
Why is TravisCi failing? Please look into it?

File: src/tests/mockData.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const users = {
+  success: {
+    email: 'Kamana@gmail.com',
+    password: 'kamana1234'
+  },
```
Comment (Mnickii) [inline src/tests/mockData.js]: Instead of mockData, let's look into seeding the database with data before we run the tests

**Review by Mnickii — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 2]
```diff
@@ -16,6 +15,7 @@ const port = process.env.PORT || 3000;
 
 app.use('/api', welcome);
 app.use('/api-doc', swagger);
+app.use('/api/v1/auth', authRouter);
```
Comment (Mnickii) [inline src/index.js:27]: duplicate

**Review by Mnickii — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (Baraka-Mugisha) [reply src/index.js:27]: it is resolved now

**Review by Mnickii — DISMISSED** [review]

**Review by izzett222 — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

**Review by Mnickii — APPROVED** [review]

**Pull request merged**